### ISSUE TITLE
Disabled jobs get scheduled and executed when upgrading

### ISSUE THREAD

**Issue by mzeis** [AOEpeople/Aoe_Scheduler#106]
When upgrading from 0.x to 1.x I noticed that jobs which were disabled were scheduled and executed:

![scheduler_disabled_cronjobs](https://cloud.githubusercontent.com/assets/371060/6730922/f416f69a-ce40-11e4-9684-6780f8131db5.png)

These jobs had been disabled in 0.x using the functionality built-in to Aoe_Scheduler (i.e. the system configuration entry containing a comma-separated list of job codes). When we upgraded the schedule for this jobs was generated once. As you can see the subsequent schedulings worked as intended: these job's haven't been generated again since.

We didn't have to re-save the cronjob configuration etc. It seems that re-writing the job status from the old entry to the new ones works but that the initial scheduling doesn't check the status correctly. We disabled cronjobs and caching when deploying so I'm pretty sure this shouldn'e be the reason.

**Comment by LeeSaferite** [comment]
Did you run DB upgrade scripts before you re-enabled the cron jobs? The migration from the old disabled format to the new one happens in an upgrade script.

**Comment by mzeis** [comment]
Yes. We ran upgrade scripts, some indexing processes (was necessary because of another update), enabled caches and then re-enabled the cron jobs.

**Comment by LeeSaferite** [comment]
Just so I'm clear when I look into this, you have 4 disabled jobs that ended up being scheduled a few times after  the upgrade. But they also stopped scheduling themselves after the initial mistake?

**Comment by mzeis** [comment]
Exactly. I assume they have been scheduled a few times because of a single call. Our settings of how long in advance jobs are created lead to multiple entries.

I cannot rule out that the upgrade script was called more than once though. We ran the upgrade script by requesting a page rather than disabling the normal Magento behaviour and triggering the upgrade via something like n98-magerun. If multiple parallel runs of the setup script cause this behaviour it also could be an explanation.
